controller/html/index.js

```diff
@@ -0,0 +1,303 @@
+    ws.send(JSON.stringify(json));
+}
+
+function onSubmit() {
+    const input = document.getElementById('raw-input');
+
+    // Defending against invalid JSON
+    try {
+        JSON.parse(input.value);
+    } catch(err) {
+        alert('Impossible to parse input into JSON! Please correct the input string');
+        return;
+    }
+
+    output(`Sent manually: ${input.value}`, 'magenta');
+    _send(JSON.parse(input.value));
+    // TODO: do we even want to revert to default one, would not
+    //   it be better to just leave it, so user can modify it?
+    input.value = templateJSON;
+    input.focus();
+}
+
+function onCloseClick() {
+    ws.close();
+}
+
+function emulatorStart(select) {
+    const version = document.getElementById(select).value;
+    _send({
+        type: 'emulator-start',
+        version,
+    });
+    setTimeout(getBackgroundStatus, 200);
+}
+
+function emulatorWipe() {
+    _send({
+        type: 'emulator-wipe',
+    });
+}
+
+function emulatorResetDevice() {
+    _send({
+        type: 'emulator-reset-device',
+    });
+}
+
+function emulatorSetup() {
+    const input = document.getElementById('seed-input');
+    _send({
+        type: 'emulator-setup',
+        mnemonic: input.value || 'all all all all all all all all all all all all',
+        pin: '',
+        passphrase_protection: false,
+        label: 'Hello!',
+    });
+}
+
+function emulatorPressYes() {
+    _send({
+        type: 'emulator-press-yes',
+    });
+}
+
+function emulatorPressNo() {
+    _send({
+        type: 'emulator-press-no',
+    });
+}
+
+function emulatorStop() {
+    _send({
+        type: 'emulator-stop',
+    });
+    setTimeout(getBackgroundStatus, 200);
+}
+
+function bridgeStart(version) {
+    _send({
+        type: 'bridge-start',
+        version,
+    });
+    setTimeout(getBackgroundStatus, 200);
+}
+
+function bridgeStop() {
+    _send({
+        type: 'bridge-stop',
+    });
+    setTimeout(getBackgroundStatus, 200);
+}
+
+function exit() {
+    _send({
+        type: 'exit',
+    });
+}
+
+function ping() {
+    _send({
+        type: 'ping',
+    });
+}
+
+function reflectBackgroundSituationInGUI(dataObject) {
+    if ('bridge_status' in dataObject) {
+        reflectBridgeSituation(dataObject.bridge_status);
+    }
+    if ('emulator_status' in dataObject) {
+        reflectEmulatorSituation(dataObject.emulator_status);
+    }
+}
+
+function reflectBridgeSituation(status) {
+    if (status.is_running) {
+        reflectBridgeStartedInGUI(status.version);
+        writeBridgeStatus(`Running - ${status.version}`);
+    } else {
+        reflectBridgeStoppedInGUI();
+        writeBridgeStatus("Stopped");
+    }
+}
+
+function reflectBridgeStartedInGUI(version) {
+    // Can happen that bridge is already running on the background, but
+    //   was not spawned by the GUI (causing confusion)
+    if (!version) {
+        alert("Please check if there is no instance of bridge running already - please kill them.");
```

Comment:
Small rephrase suggestion: 
`It seems you already have an instance of bridge running - please kill it.");

Reply:
Agreed, this is more simple